pkg/workceptor/workceptor.go

```diff
@@ -631,6 +636,11 @@ func (w *Workceptor) scanForUnits() {
 				}
 				w.activeUnits[fi.Name()] = unit
 				if si.Node != "" && si.Node != w.nc.NodeID() {
+					if si.LocalCancel && si.State == WorkStateRunning {
```

Comment:
on control node restart, this will make sure we restart the cancellation goroutine

Reply:
I realize we need a way to determine if a previous cancel attempt timed out, and not re-try a cancel in that case. Otherwise, every time we restart the control service it will respawn the go routines to cancel previously timed out cancellations.

Reply:
What if it's WorkStatePending?  Then we don't launch the monitor, and maybe the job goes to WorkStateRunning a second later.  Maybe this should be `if si.LocalCancel && !IsComplete(si.State)`.

Reply:
changed it to `!IsComplete(si.State)`